libpod/container.go

```diff
@@ -407,8 +407,54 @@ func (c *Container) teardownStorage() error {
 	return nil
 }
 
+// Refresh refreshes the container's state after a restart
+func (c *Container) refresh() error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if !c.valid {
+		return errors.Wrapf(ErrCtrRemoved, "container %s is not valid - may have been removed", c.ID())
+	}
+
+	// We need to get the container's temporary directory from c/storage
+	// It was lost in the reboot and must be recreated
+	dir, err := c.runtime.storageService.GetRunDir(c.ID())
+	if err != nil {
+		return errors.Wrapf(err, "error retrieving temporary directory for container %s", c.ID())
+	}
+	c.state.RunDir = dir
+
+	// The container is no longer mounted
+	c.state.Mounted = false
+	c.state.Mountpoint = ""
+
+	// The container is no longe running
+	c.state.PID = 0
+
+	// Check the container's state. If it's not created in runc yet, we're
+	// done
+	if c.state.State == ContainerStateConfigured {
+		if err := c.runtime.state.SaveContainer(c); err != nil {
+			return errors.Wrapf(err, "error refreshing state for container %s", c.ID())
+		}
+
+		return nil
+	}
+
+	// The container must be recreated in runc
+	if err := c.init(); err != nil {
```

Comment:
I don't think we should do this.  This should happen on kpod start.

Reply:
That is fine for kpod, but what about CRI-O? Do we want to ping Init() every time we create a container there?
I feel like requiring repeated invocations of Init() is really bad API design. I almost thing we should get rid of it, and just do it automatically when the user makes an API call that would require the container being in runc.

Reply:
I don't think CRI-O ever uses an existing container.  It is always doing a create/start.  When a kubenode reboots, it always recreates all containers, which might lead us to a different issue where we end up with a lot of abandoned containers.